src/app/public/modules/shared/config.extension.ts

```diff
@@ -0,0 +1,8 @@
+import {Injectable} from '@angular/core';
+import {HelpWidgetConfig} from './widget-config';
+import {Observable} from 'rxjs';
+
+@Injectable()
+export abstract class ConfigExtension {
```

Comment:
I think you could interpret the scenario as actually needing an implementation of this class - where this class' `extends` method returns `of(config)` instead of having to check for undefined and perform it on [line 52](https://github.com/blackbaud/skyux-lib-help/pull/57/files#diff-188096fa615cef180f1028f6312854bbR52).

That pattern is how the UI config service works.

- https://github.com/blackbaud/skyux-core/blob/master/src/app/public/modules/ui-config/ui-config.service.ts
- https://docs.blackbaud.com/ui-config-service-docs/learn/getting-started

I think both patterns could work, but the latter feels a little cleaner to me.

It also feels a little weird not having `Service` in the classes name since it is using Angular's DI.  Perhaps `HelpInitializationConfigService` and consumers could provide `AppHelpInitializationConfigService`?